docs/src/content/docs/reference/imports.md

```diff
@@ -0,0 +1,230 @@
+
+```aw
+# Imported file
+---
+tools:
+  github:
+    allowed: [get_issue, list_issues]
+---
+```
+
+```aw
+# Main workflow
+---
+imports:
+  - org/repo v1.0 tools-config.md
+tools:
+  github:
+    allowed: [get_pull_request]
+---
+```
+
+Result: GitHub tools will have both sets of allowed functions.
+
+**Other fields**: Imported values take precedence for non-tools fields.
+
+## Version Control
+
+The `gh aw install` command automatically creates a `.aw/.gitignore` file that ignores the `imports/` folder. This ensures the cached imported files are not committed while the lock file remains trackable.
+
+You can also add `.aw/imports/` to your project's root `.gitignore` for additional protection:
+
+```gitignore
+# Workflow imports (cached files only, not the lock file)
+.aw/imports/
+```
+
+The lock file (`.aw/imports.lock`) should be committed to version control to ensure reproducible builds by pinning exact commit SHAs, similar to `go.sum` or `package-lock.json`.
+
+## Best Practices
+
+1. **Use version tags**: Prefer semantic version tags (v1.0.0) over branch names for stability
+2. **Install before compile**: Always run `gh aw install` after adding or updating imports
+3. **Commit lock file**: Include `.aw/imports.lock` in version control for reproducibility
+4. **Test imports**: Verify imported content compiles successfully
+5. **Document dependencies**: Document why each import is needed
+
+## Transitive Dependencies
+
+Imported files can contain @include directives. The install command automatically:
+- Discovers transitive @include references
+- Validates all referenced files exist
+- Records them in the lock file
+
+## Comparison with @include
+
+| Feature | @include | imports |
+|---------|----------|---------|
+| Source | Local repository | External repositories |
+| Resolution | Compile time | Install time |
+| Caching | None (always read) | Cached in .aw/imports/ |
+| Version control | Implicit (via git) | Explicit (via lock file) |
+| Use case | Project-local reuse | Cross-repo sharing |
+
+Use @include for local files, imports for external dependencies.
+
+## Troubleshooting
+
+**Import not found during compile**:
+```
+error: import org/repo version path not found in lock file
+```
+Solution: Run `gh aw install` to install imports.
+
+**Version resolution failed**:
+```
+error: failed to resolve version v1.0.0 to commit SHA
+```
+Solution: Verify the version/tag/branch exists in the repository.
+
+**File not found in imported repository**:
+```
+error: imported file not found: path/to/file.md
+```
+Solution: Check the file path exists in the repository at the specified version.
+
+## Example: Importing Security Notices
+
+```aw
+---
+on:
+  issues:
+    types: [opened]
+permissions:
+  contents: read
+  issues: write
+engine: claude
+imports:
+  - githubnext/gh-aw main .github/workflows/agentics/shared/tool-refused.md
```

Comment:
In GitHub Actions `uses` the syntax is simpler, like this:

`githubnext/gh-aw/.github/workflows/agentics/shared/tool-refused.md@main`

e.g.

```
steps:
- name: Start the Repo-mind MCP server
  id: repo-mind-server
  uses: githubnext/repo-mind/.github/actions/server@main
```

We should align with this I think